activerecord/lib/active_record/associations/builder/belongs_to.rb

```diff
@@ -123,7 +123,17 @@ def self.define_validations(model, reflection)
       super
 
       if required
-        model.validates_presence_of reflection.name, message: :required
+        condition = lambda { |record|
+          foreign_key = reflection.foreign_key
+          foreign_type = reflection.foreign_type
+
+          record.read_attribute(foreign_key).nil? ||
+            record.attribute_changed?(foreign_key) ||
+            (reflection.polymorphic? && (record.read_attribute(foreign_type).nil? || record.attribute_changed?(foreign_type))) ||
+            !ActiveRecord.belongs_to_required_validates_foreign_key
```

Comment:
We can assume this one is set during boot and won't change. So if it's set, we can just not generate a condition at all.

Reply:
Also I think the naming suggest that it should be used the other way around:

  - if `belongs_to_required_validates_foreign_key  == true` then we just put no condition, we always validates the presence.
  - if `belongs_to_required_validates_foreign_key  == false` then we only validate presence when the foreign key changed.

Reply:
Thanks! Under `belongs_to_required_validates_foreign_key` I meant something like "validates foreign key attribute for presence, not the association". But yeah, your interpretation is correct. So flipped it.